Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -311,27 +311,34 @@ open class RouteController: NSObject {
     }
 
     private func update(to status: NavigationStatus) {
-        guard let location = rawLocation else { return }
+        guard let rawLocation = rawLocation else { return }
+        
+        let snappedLocation = CLLocation(status.location)
+        
         // Notify observers if the step’s remaining distance has changed.
-        update(progress: routeProgress, with: CLLocation(status.location), rawLocation: location, upcomingRouteAlerts: status.upcomingRouteAlerts)
+        update(progress: routeProgress,
+               with: snappedLocation,
+               rawLocation: rawLocation,
+               upcomingRouteAlerts: status.upcomingRouteAlerts)
         
         updateIndexes(status: status, progress: routeProgress)
         updateRouteLegProgress(status: status)
-        let willReroute = !userIsOnRoute(location, status: status) && delegate?.router(self, shouldRerouteFrom: location)
-            ?? DefaultBehavior.shouldRerouteFromLocation
+        let willReroute = !userIsOnRoute(snappedLocation, status: status)
+        && delegate?.router(self, shouldRerouteFrom: snappedLocation)
+        ?? DefaultBehavior.shouldRerouteFromLocation
```

Comment:
Could you place brackets here as it's difficult to visually analyze this part? I.e.
```suggestion
        && (delegate?.router(self, shouldRerouteFrom: snappedLocation)
            ?? DefaultBehavior.shouldRerouteFromLocation)
```